projects/i18n/tools/extract-i18n.ts

```diff
@@ -1,18 +1,24 @@
 import {inject} from '@angular/core';
-import {Language} from '@taiga-ui/i18n/interfaces';
-import {Observable} from 'rxjs';
-import {map} from 'rxjs/operators';
+import {TuiLanguage} from '@taiga-ui/i18n/interfaces';
+import {isObservable, Observable, of} from 'rxjs';
+import {map, switchMap} from 'rxjs/operators';
 
 import {TUI_LANGUAGE} from './language';
 
 /**
  * @deprecated: use {@link tuiExtractI18n} instead
  */
 // eslint-disable-next-line @typescript-eslint/naming-convention
-export function extractI18n<K extends keyof Language>(
+export function extractI18n<K extends keyof TuiLanguage>(
     key: K,
-): () => Observable<Language[K]> {
-    return () => inject(TUI_LANGUAGE).pipe(map(lang => lang[key]));
+): () => Observable<TuiLanguage[K]> {
+    return () =>
+        inject(TUI_LANGUAGE).pipe(
+            switchMap((streamOrValue: Observable<TuiLanguage> | TuiLanguage) =>
```

Comment:
Why the type change of `TUI_LANGUAGE`?

Reply:
Because we need to set asynchronous data
 
<img width="787" alt="image" src="https://user-images.githubusercontent.com/12021443/182424945-a027308f-84cb-4728-a17a-db6fb56b0da1.png">